packages/core-data/src/selectors.js

```diff
@@ -231,7 +231,7 @@ export const __experimentalGetDirtyEntityRecords = createSelector(
 									entity.key || DEFAULT_ENTITY_KEY
 								],
 							title: ! entity.getTitle
-								? ''
+								? entityRecord.title || ''
```

Comment:
what if we instead implement the `entity.getTitle` for the "site" entity in the "entities" config instead?

Reply:
here I mean https://github.com/WordPress/gutenberg/blob/master/packages/core-data/src/entities.js#L20-L25

Reply:
That would be great if we could implement the `getTitle` for this instead.  However, I'm not entirely certain how to go about it.

From the default entities declaration as you posted in the link above, the postType doesn't have this added there either: https://github.com/WordPress/gutenberg/blob/0b3f6ef1efaa1f15afeda8065f1ba2c2d4279fda/packages/core-data/src/entities.js#L26-L32

but in a `loadPostTypeEntities` promise that the site entity doesnt seem to have a parallel for here: 
https://github.com/WordPress/gutenberg/blob/0b3f6ef1efaa1f15afeda8065f1ba2c2d4279fda/packages/core-data/src/entities.js#L117-L122

🤔

Reply:
I believe just adding the getTitle function to that object works. 

The difference with posts is that we have this concept of "meta entities" (or entities that become "kinds" of other entities), that's what happens with "post types". We can't just have a fixed list of CPT entities as these are dynamic so we generate them. but the site entity is static.

Reply:
Also, I'm not certain why the site entity has `title` already implemented where the postTypes don't and use the `getTitle` function instead.  Since `title` is already implemented on the site entity, would adding `getTitle` be redundant?

Reply:
`getTitle` seems a more "generic" to get a "display label" for an entity regardless of its object format. If we default to the "title" property, I'm not sure it would work properly for all entities.

Reply:
Oh cool, that makes sense.  Thank you.

I added a `getTitle` function to the site entity object:
```
	getTitle: ( record ) => {
		return get( record, [ 'title' ], __( 'Site Title' ) );
	},
```
and that seems to work well. =)